crud/compare/comparators.lua

```diff
@@ -150,12 +154,23 @@ function comparators.gen_tuples_comparator(cmp_operator, key_parts, field_names,
     local updated_key_parts = comparators.update_key_parts_by_field_names(
             space_format, field_names, key_parts
     )
+
     local keys_comparator = comparators.gen_func(cmp_operator, updated_key_parts)
 
+    local key_def
+    if has_keydef then
+        key_def = keydef_lib.new(updated_key_parts)
+    end
```

Comment:
```lua
if has_keydef then
    local key_def = keydef_lib.new(updated_key_parts)
    return function(lhs, rhs)
          local lhs_key = key_def:extract_key(lhs)
          local rhs_key = key_def:extract_key(rhs)
          return keys_comparator(lhs_key, rhs_key)
    end
else
    return function(lhs, rhs)
        local lhs_key = utils.extract_key(lhs, updated_key_parts)
        local rhs_key = utils.extract_key(rhs, updated_key_parts)
        return keys_comparator(lhs_key, rhs_key)
    end
end
```

I think it looks better and eliminates runtime `has_keydef` flag check inside comparator.